docs/hub/datasets-pyarrow.md

```diff
@@ -0,0 +1,234 @@
+
+<Tip>
+
+Content defined chunking (CDC) makes the Parquet writer chunk the data pages in a way that makes duplicate data chunked and compressed identically.
+Without CDC, the pages are arbitrarily chunked and therefore duplicate data are impossible to detect because of compression.
+Thanks to CDC, Parquet uploads and downloads from Hugging Face are faster, since duplicate data are uploaded or downloaded only once.
+
+</Tip>
+
+Find more information about Xet [here](https://huggingface.co/join/xet).
+
+## Use Images
+
+You can load a folder with a metadata file containing a field for the names or paths to the images, structured like this:
+
+```
+Example 1:            Example 2:
+folder/               folder/
+├── metadata.parquet  ├── metadata.parquet
+├── img000.png        └── images
+├── img001.png            ├── img000.png
+...                       ...
+└── imgNNN.png            └── imgNNN.png
+```
+
+You can iterate on the images paths like this:
+
+```python
+from pathlib import Path
+import pyarrow as pq
+
+folder_path = Path("path/to/folder")
+table = pq.read_table(folder_path + "metadata.parquet")
+for file_name in table["file_name"].to_pylist():
+    image_path = folder_path / file_name
+    ...
+```
+
+Since the dataset is in a [supported structure](https://huggingface.co/docs/hub/en/datasets-image#additional-columns) (a `metadata.parquet` file with a `file_name` field), you can save this dataset to Hugging Face and the Dataset Viewer shows both the metadata and images on Hugging Face.
+
+```python
+from huggingface_hub import HfApi
+api = HfApi()
+
+api.upload_folder(
+    folder_path=folder_path,
+    repo_id="username/my_image_dataset",
+    repo_type="dataset",
+)
+```
+
+### Embed Images inside Parquet
+
+PyArrow has a binary type which allows to have the images bytes in Arrow tables. Therefore it enables saving the dataset as one single Parquet file containing both the images (bytes and path) and the samples metadata:
+
+```python
+import pyarrow as pa
+import pyarrow.parquet as pq
+
+# Embed the image bytes in Arrow
+image_array = pa.array([
+    {
+        "bytes": (folder_path / file_name).read_bytes(),
+        "path": file_name,
+    }
+    for file_name in table["file_name"].to_pylist()
+])
+table.append_column("image", image_array)
+
+# (Optional) Set the HF Image type for the Dataset Viewer and the `datasets` library
+features = {"image": {"_type": "Image"}}  # or using datasets.Features(...).to_dict()
+schema_metadata = {"huggingface": {"dataset_info": {"features": features}}}
+table = table.replace_schema_metadata(schema_metadata)
+
+# Save to Parquet
+# (Optional) with use_content_defined_chunking for faster uploads and downloads
+# (Optional) with row_group_size to allow loading 100 images at a time
+pq.write_table(table, "data.parquet", use_content_defined_chunking=True, row_group_size=100)
```

Comment:
We shouldn't recommend small `row_group_size` values because such a small value effectively disables the benefits of parquet CDC.